torch/_higher_order_ops/while_loop.py

```diff
@@ -21,31 +21,40 @@
 
 
 class WhileLoopOp(HigherOrderOperator):
-    def __call__(self, cond_fn, body_fn, operands):
+    def __call__(self, cond_fn, body_fn, carried_inputs, additional_inputs):
         if not isinstance(cond_fn, torch.fx.GraphModule) or not isinstance(
             body_fn, torch.fx.GraphModule
         ):
             raise RuntimeError(
                 "cond_fn and body_fn must be torch.fx.GraphModule, got "
                 f"{type(cond_fn)} and {type(body_fn)}"
             )
-        if not isinstance(operands, tuple):
-            raise RuntimeError("operands must be a tuple, got " f"{type(operands)}")
-        if not all(isinstance(t, (torch.Tensor, int, float, bool)) for t in operands):
+        if not isinstance(carried_inputs, tuple):
             raise RuntimeError(
-                "operands must be a tuple of tensors, ints, floats, or bools, got "
-                f"{operands}"
+                "carried_inputs must be a tuple, got " f"{type(carried_inputs)}"
             )
-        return super().__call__(cond_fn, body_fn, operands)
+        if not isinstance(additional_inputs, tuple):
+            raise RuntimeError(
+                "additional_inputs must be a tuple, got " f"{type(additional_inputs)}"
```

Comment:
UX question: if `additional_inputs` are generated internally (and are *not* considered a part of public API), would the errors re. `additional_inputs` make sense to the user?